src/hooks/common/useDebounce.ts

```diff
@@ -0,0 +1,25 @@
+import { useRef } from 'react';
+
+/**
+ * 함수에 debounce를 적용하는 hook입니다. ms 이내의 실행을 병합하여 한 번 실행합니다.
+ *
+ * @param cb debounce 적용을 원하는 함수
+ * @param ms debounce timeout
+ * @returns debouncedFn
+ */
+const useDebounce = (cb: () => void, ms: number): (() => void) => {
+  const timer = useRef<ReturnType<typeof setTimeout>>();
+
+  const dispatchDebounce = () => {
+    if (timer.current) {
+      clearTimeout(timer.current);
+    }
+    const newTimer = setTimeout(() => {
+      cb();
+    }, ms);
+    timer.current = newTimer;
```

Comment:
```suggestion
    const newTimer = setTimeout(cb, ms);
    timer.current = newTimer;
```

이렇게 해도 될 거 같은데 뇌피셜이긴 합니다

지금도 좋아요 💯 👍

Reply:
당연히 될줄알고 밀어넣었는데 안되네요...? 왜지...????
[39cc22f](https://github.com/depromeet/na-lab-client/pull/40/commits/39cc22f6c4246a6f3d9b222dec99a52f8cec9eaf)에서 원복했습니다

Reply:
왜안되긴... 잘못작성했었네요 [51b296a](https://github.com/depromeet/na-lab-client/pull/40/commits/51b296a5d8a4211781b31638028b5a59752179a8)에서 화살표함수 없애는방식으로 수정했습니다